rust/agama-network/src/model.rs

```diff
@@ -0,0 +1,439 @@
+            ..Default::default()
+        };
+        let conn0 = Connection::Ethernet(EthernetConnection { base: base0 });
+        state.add_connection(conn0).unwrap();
+        state.remove_connection(uuid).unwrap();
+        let found = state.get_connection(uuid).unwrap();
+        assert!(found.is_missing());
+    }
+
+    #[test]
+    fn test_remove_unknown_connection() {
+        let mut state = NetworkState::default();
+        let error = state.remove_connection(Uuid::new_v4()).unwrap_err();
+        assert!(matches!(error, NetworkStateError::UnknownConnection(_)));
+    }
+}
+
+/// Network device
+#[derive(Debug, Clone)]
+pub struct Device {
+    pub name: String,
+    pub ty: DeviceType,
+}
+
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub enum DeviceType {
+    Loopback = 0,
+    Ethernet = 1,
+    Wireless = 2,
+    Unknown = 3,
+}
+
+impl TryFrom<u8> for DeviceType {
+    type Error = NetworkStateError;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(DeviceType::Loopback),
+            1 => Ok(DeviceType::Ethernet),
+            2 => Ok(DeviceType::Wireless),
+            3 => Ok(DeviceType::Unknown),
+            _ => Err(NetworkStateError::InvalidDeviceType(value)),
+        }
+    }
+}
+
+/// Represents an available network connection
+#[derive(Debug, PartialEq, Clone)]
+pub enum Connection {
+    Ethernet(EthernetConnection),
+    Wireless(WirelessConnection),
+    Loopback(LoopbackConnection),
+}
+
+impl Connection {
+    pub fn new(id: String, device_type: DeviceType) -> Self {
+        let base = BaseConnection {
+            id: id.to_string(),
+            ..Default::default()
+        };
+        match device_type {
+            DeviceType::Wireless => Connection::Wireless(WirelessConnection {
+                base,
+                ..Default::default()
+            }),
+            _ => Connection::Ethernet(EthernetConnection { base }),
+        }
+    }
+
+    pub fn base(&self) -> &BaseConnection {
+        match &self {
+            Connection::Ethernet(conn) => &conn.base,
+            Connection::Wireless(conn) => &conn.base,
+            Connection::Loopback(conn) => &conn.base,
+        }
+    }
+
+    pub fn base_mut(&mut self) -> &mut BaseConnection {
+        match self {
+            Connection::Ethernet(conn) => &mut conn.base,
+            Connection::Wireless(conn) => &mut conn.base,
+            Connection::Loopback(conn) => &mut conn.base,
+        }
+    }
+
+    pub fn id(&self) -> &str {
+        self.base().id.as_str()
+    }
+
+    pub fn uuid(&self) -> Uuid {
+        self.base().uuid
+    }
+
+    pub fn ipv4(&self) -> &Ipv4Config {
+        &self.base().ipv4
+    }
+
+    pub fn ipv4_mut(&mut self) -> &mut Ipv4Config {
+        &mut self.base_mut().ipv4
+    }
+
+    pub fn remove(&mut self) {
+        self.base_mut().status = Status::Missing;
+    }
+
+    pub fn is_missing(&self) -> bool {
+        self.base().status == Status::Missing
```

Comment:
this is kind of inconsistent with action and is_action being different. What about `dismiss` and `is_missing` or `remove` and `is_removed`?

Reply:
with is_action I mean action like action => add so method `add` and `is_added`. So inconsistency for me is different name of actions. is_removed sounds good to me.